src/Database/DatabaseServiceProvider.php

```diff
@@ -63,5 +75,35 @@ public function boot()
     {
         AbstractModel::setConnectionResolver($this->app->make(ConnectionResolverInterface::class));
         AbstractModel::setEventDispatcher($this->app->make('events'));
+
+        foreach ($this->app->make('flarum.model.customRelations') as $modelClass => $callbacks) {
+            foreach ($callbacks as  $name => $relation) {
+                if (is_string($relation)) {
+                    $relation = $this->app->make($relation);
+                }
+
+                AbstractModel::addCustomRelation($modelClass, $name, $relation);
+            }
+        }
+
+        foreach ($this->app->make('flarum.model.dateCallbacks') as $modelClass => $callbacks) {
+            foreach ($callbacks as $callback) {
+                if (is_string($callback)) {
+                    $callback = $this->app->make($callback);
+                }
+
+                AbstractModel::addDateCallback($modelClass, $callback);
+            }
```

Comment:
If we decide to switch to the non-callback solution which I proposed in the other comment, this could just assign directly to the (public) property on the model.

I don't think we should allow injection from the container into the callbacks (this would only affect default attributes, as per my proposal), that seems like overkill for this use case.

Reply:
Agreed

Reply:
All are done